src/components/createPostPopup/index.tsx

```diff
@@ -1,56 +1,86 @@
-import { useState } from "react";
+import { useState, FunctionComponent } from "react";
+import { Modal, Button } from "react-bootstrap";
+import { Room, RoomUser } from "@prisma/client";
+
 import SurveyPost from "./surveyPost";
 import TextPost from "./textPost";
 
 interface PostProps {
   showTextPost: boolean;
+  room: Room;
+}
+
+interface CreateProps {
+  onHide: () => void;
+  show: boolean;
+  room: Room & { users: RoomUser[] };
 }
 
-function Post(props: PostProps) {
+function Post(props: PostProps): JSX.Element {
   if (props.showTextPost == false) {
-    return <SurveyPost />;
+    return <SurveyPost room={props.room} />;
   } else {
-    return <TextPost />;
+    return <TextPost room={props.room} />;
   }
 }
 
-const CreatePost = () => {
-  const [showTextPost, setTextPost] = useState(false);
+const CreatePost: FunctionComponent<CreateProps> = (props: CreateProps) => {
+  const [showTextPost, setTextPost] = useState(true);
 
-  const openSurveyPost = () => {
+  const openSurveyPost = (): void => {
     setTextPost(false);
   };
 
-  const openTextPost = () => {
+  const openTextPost = (): void => {
     setTextPost(true);
   };
 
   return (
-    <div className="bg-dark vh-100">
-      <div>
-        <h2 className="text-primary">Create a Post!</h2>
-        <div>
-          <button
-            className="btn btn-primary m-1"
-            type="button"
-            onClick={openSurveyPost}
-          >
-            Survey Post
-          </button>
-          <button
-            className="btn btn-primary m-1"
-            type="button"
-            onClick={openTextPost}
-          >
-            Text Post
-          </button>
-        </div>
-      </div>
-
-      <div>
-        <Post showTextPost={showTextPost}></Post>
+    <Modal
+      {...props}
+      size="lg"
+      aria-labelledby="contained-modal-title-vcenter"
+      centered
+      className="my-modal"
+    >
+      <div className="bg-dark rounded">
+        <Modal.Header
+          closeButton
+          closeVariant="white"
+          className="modal-footer border-0"
+        >
+          <div>
+            <h2 className="text-primary">
+              Create a Post in {props.room.name}!
+            </h2>
+            <div>
+              <button
+                className="btn btn-primary m-1"
+                type="button"
+                onClick={openTextPost}
+              >
+                Text Post
+              </button>
+              <button
+                className="btn btn-primary m-1"
+                type="button"
+                onClick={openSurveyPost}
+              >
+                Survey Post
+              </button>
```

Comment:
```suggestion
              <Button
                className="m-1"
                type="button"
                onClick={openTextPost}
              >
                Text Post
              </Button>
              <Button
                className="m-1"
                type="button"
                onClick={openSurveyPost}
              >
                Survey Post
              </Button>
```
Umwandlung zu `react-bootstrap`